src/charm.py

```diff
@@ -542,11 +545,11 @@ def _alerting_config(self) -> dict:
         alerting_config = {"alertmanagers": [{"static_configs": [{"targets": alertmanagers}]}]}
         return alerting_config
 
-    def _prometheus_config(self) -> str:
+    def _prometheus_config(self) -> Tuple[str, dict]:
```

Comment:
This return type doesn't seem logically consistent or Pythonic. It isn't returning "just" the Prometheus config.

Considering that the only thing done with the returned certs is pushing them to the container, it would probably make more sense to refactor `_prometheus_config` to `_generate_prometheus_config` or similar and handle the writing there, so "get the cert and write it to disk" are logically in the same place, especially since we don't do any other processing or handling of it whatsover

Reply:
So `self._prometheus_config()` is only "converting" reldata to useful stuff, and then both the config yaml and the certs are written together. Kind of separation of concerns. From first glance it seems that this would have a lower spaghetti factor.

I agree it's confusing the "config" on the left hand side means the yaml, but on the right hand side it's the "entire config":

`prometheus_config, certs = self._prometheus_config()`

Reply:
Oh you mean move everything into the same function. Sure, that's probably better.